src/main/java/org/ecocean/Occurrence.java

```diff
@@ -765,53 +765,53 @@ public JSONObject uiJson(HttpServletRequest request) throws JSONException {
 
     JSONObject encounterInfo = new JSONObject();
     for (Encounter enc : this.encounters) {
-      encounterInfo.put(enc.getOccurrenceID(), new JSONObject("{url: "+enc.getUrl(request)+"}"));
+      JSONObject urlInfo = new JSONObject();
+      urlInfo.put("url", enc.getUrl(request));
+      if(enc.getOccurrenceID()!=null) encounterInfo.put(enc.getOccurrenceID(), urlInfo);
     }
     jobj.put("encounters", encounterInfo);
     jobj.put("assets", this.assets);
 
     jobj.put("groupBehavior", this.getGroupBehavior());
-    return jobj;
-
-  }
-
-/*  this was messing up the co-occur js (d3?), so lets kill for now?
-  public org.datanucleus.api.rest.orgjson.JSONObject sanitizeJson(HttpServletRequest request,
-                org.datanucleus.api.rest.orgjson.JSONObject jobj) throws org.datanucleus.api.rest.orgjson.JSONException {
-            return sanitizeJson(request, jobj, true);
-        }
-
-  public org.datanucleus.api.rest.orgjson.JSONObject sanitizeJson(HttpServletRequest request, org.datanucleus.api.rest.orgjson.JSONObject jobj, boolean fullAccess) throws org.datanucleus.api.rest.orgjson.JSONException {
-    jobj.put("ID", this.ID);
-    jobj.put("encounters", this.encounters);
-    if ((this.getEncounters() != null) && (this.getEncounters().size() > 0)) {
-        JSONArray jarr = new JSONArray();
-  ///  *if* we want full-blown:  public JSONObject Encounter.sanitizeJson(HttpServletRequest request, JSONObject jobj) throws JSONException {
-        //but for *now* (see note way above) this is all we need for gallery/image display js:
-        for (Encounter enc : this.getEncounters()) {
-            JSONObject je = new JSONObject();
-            je.put("id", enc.getID());
-            if (enc.hasMarkedIndividual()) je.put("individualID", enc.getIndividualID());
-            if ((enc.getAnnotations() != null) && (enc.getAnnotations().size() > 0)) {
-                JSONArray ja = new JSONArray();
-                for (Annotation ann : enc.getAnnotations()) {
-                    ja.put(ann.getId());
-                }
-                je.put("annotations", ja);
-            }
-            jarr.put(je);
-        }
-        jobj.put("encounters", jarr);
-    }
-    int[] assetIds = new int[this.assets.size()];
-    for (int i=0; i<this.assets.size(); i++) {
-      if (this.assets.get(i)!=null) assetIds[i] = this.assets.get(i).getId();
-    }
-    jobj.put("assets", assetIds);
-    return jobj;
-
-  }
-*/
+    return sanitizeJson(request, decorateJson(request, jobj));
+  }
+
+// /*  this was messing up the co-occur js (d3?), so lets kill for now?
```

Comment:
not sure why this is sorta _double_-commented out!  but going to not really care about it.  :smile: